### DISCUSSION TITLE
How to add password for printing?

### DISCUSSION THREAD

**Discussion by nissansz** [pymupdf/PyMuPDF#2038] [Q&A] [answered]
How to add password for printing?

**Comment by JorjMcKie** [comment, marked as answer]
Please look at the "permissions" bit field (an integer) [here](https://pymupdf.readthedocs.io/en/latest/recipes-general.html#how-to-deal-with-pdf-encryption).

**Comment by nissansz** [comment]
perm = int(
    fitz.PDF_PERM_ACCESSIBILITY  # always use this
    # | fitz.PDF_PERM_PRINT  # permit printing
    # | fitz.PDF_PERM_COPY  # permit copying
    # | fitz.PDF_PERM_ANNOTATE  # permit annotations
)
owner_pass = "owner"  # owner password
user_pass = "user"  # user password
encrypt_meth = fitz.PDF_ENCRYPT_AES_256  # strongest algorithm

I can see that owner can have all permissions. User cannot print, any option to let user to input a password for printing?

**Reply by JorjMcKie** [reply]
> any option to let user to input a password for printing?

No. PDF only supports one owner and one user password.

**Reply by JorjMcKie** [reply]
This is **not** a MuPDF restriction.

**Comment by nissansz** [comment]
Understood
